pkg/ingester/ingester_test.go

```diff
@@ -6493,13 +6723,159 @@ func TestIngesterTruncationOfEphemeralSeries(t *testing.T) {
 	_, err = i.Push(ctx, req)
 	require.NoError(t, err)
 
+	// Query back the series and verify that sample exists.
+	verifySamples := func(t *testing.T, expected model.Matrix) {
+		s := &stream{ctx: ctx}
+		err = i.QueryStream(&client.QueryRequest{
+			StartTimestampMs: math.MinInt64,
+			EndTimestampMs:   math.MaxInt64,
+			Matchers: []*client.LabelMatcher{
+				// Query everything from ephemeral storage.
+				{Type: client.REGEX_MATCH, Name: labels.MetricName, Value: ".*"},
+				{Type: client.EQUAL, Name: StorageLabelName, Value: EphemeralStorageLabelValue},
+			},
+		}, s)
+		require.NoError(t, err)
+
+		res, err := chunkcompat.StreamsToMatrix(model.Earliest, model.Latest, s.responses)
+		require.NoError(t, err)
+		if len(res) == 0 {
+			res = nil
+		}
+		assert.Equal(t, expected, res)
+	}
+
+	verifySamples(t, model.Matrix{
+		&model.SampleStream{
+			Metric: map[model.LabelName]model.LabelValue{labels.MetricName: "test"},
+			Values: []model.SamplePair{
+				{Value: 100, Timestamp: model.Time(now.Add(-9 * time.Minute).UnixMilli())},
+			},
+		},
+	})
+
 	db := i.getTSDB(userID)
 	require.NotNil(t, db)
 
 	// Advance time for ephemeral storage
 	require.Nil(t, db.Compact(now.Add(5*time.Minute)))
 
+	// Old sample is no longer available for querying.
+	verifySamples(t, nil)
+
 	// Pushing the same request should now fail, because min valid time for ephemeral storage has moved on to (now + 5 minutes - ephemeral series retention = now - 5 minutes)
 	_, err = i.Push(ctx, req)
 	require.Equal(t, err, httpgrpc.Errorf(http.StatusBadRequest, wrapWithUser(newIngestErrSampleTimestampTooOld(model.Time(req.EphemeralTimeseries[0].Samples[0].TimestampMs), metricLabels), userID).Error()))
 }
+
+func TestIngesterQueryingWithStorageLabel(t *testing.T) {
```

Comment:
The test's name sounds like it is a relatively generic test which checks whether the querying with the storage label works, but actually it seems to focus just on whether the correct errors get returned.

```suggestion
func TestIngesterQueryingWithStorageLabelErrorHandling(t *testing.T) {
```